### DISCUSSION TITLE
Feature Request: Create launchSettings.json in new projects with ASPNETCORE_ENVIRONMENT Local

### DISCUSSION THREAD

**Discussion by jamiepollock** [umbraco/Umbraco.Cloud.Issues#812] [Ideas]
### Issue description

When creating a new Umbraco Cloud instance projects start with a launchSettings.json that has the ASPNETCORE_ENVIRONMENT set to Development. This is pretty standard practice however when a project grows and a Development environment is added Umbraco Cloud uses "Development" for this environment too. So any local specific settings for this environment are also reused for the remote Development environment too.

Wouldn't it make more sense to start off on the right foot and include "Local" by default as launchSettings.json is only ever used for local development?

This is something that we change on our projects right away but it might be a nice quality of life feature for new projects.

**Comment by jamiepollock** [comment]
Also didn't mention this but this would also require an appsettings.Local.json is added too.

**Comment by ronaldbarendse** [comment]
Thanks for the suggestion, and apologies for the long silence on this one. There is now a change in the CMS that addresses the underlying problem, although in a slightly different way than proposed here.

With [umbraco/Umbraco-CMS#22817](https://github.com/umbraco/Umbraco-CMS/pull/22817) (shipping in Umbraco 17.7 and 18.2, both currently available as release candidates) the project template gets a machine-local settings file instead of a new environment name:

- `Program.cs` in the template now registers `appsettings.Local.json` as the last configuration source, wrapped in `#if DEBUG`. It is loaded when you run or debug locally and is compiled out of Release builds.
- The template's `.gitignore` excludes `appsettings.Local.json`, and the project file excludes it from publish output, so it never leaves your machine.
- For projects that are not on Umbraco Cloud, the installer now also writes the database connection string to that file instead of `appsettings.json`. Cloud projects already get a local SQLite connection string generated automatically, so nothing changes there.

The result is the separation you were after: anything that should only apply to your own machine goes in `appsettings.Local.json`, while `appsettings.Development.json` keeps its role as the committed configuration for the Development environment, whether that is local or on Umbraco Cloud. Keeping `ASPNETCORE_ENVIRONMENT` set to `Development` also means `IsDevelopment()` checks, the developer exception page and other tooling that keys off that value keep working as before.

As a side note, the CMS has technically supported `appsettings.Local.json` for a while via `ConfigureUmbracoDefaults`, but because that code lives in the NuGet package (compiled in Release) the `#if DEBUG` block was stripped and the file was never actually loaded in customer projects. Moving the registration into the template's `Program.cs` fixes that, since your own build configuration now decides whether it is included.

Umbraco Cloud scaffolds new projects from the same `dotnet new umbraco` template, so projects created on Cloud after the 17.7 and 18.2 bundles are released will have this out of the box. For existing projects you can get the same behaviour by adding the `AddJsonFile` line to your `Program.cs` and the file to your `.gitignore`.
